internal/tools/semconv-gen/generator.go

```diff
@@ -0,0 +1,300 @@
+	if err != nil {
+		panic(err)
+	}
+}
+
+type config struct {
+	input    string
+	output   string
+	target   string
+	template string
+	image    string
+}
+
+func buildConfig() (*config, error) {
+	if *inputPath == "" {
+		return nil, errors.New("input path must be provided")
+	}
+
+	if *outputFn == "" {
+		*outputFn = fmt.Sprintf("%s.go", path.Base(*inputPath))
+	}
+
+	if !path.IsAbs(*outputPath) {
+		root, err := findRepoRoot()
+		if err != nil {
+			return nil, err
+		}
+		*outputPath = path.Join(root, *outputPath)
+	}
+
+	if !path.IsAbs(*templateFn) {
+		pwd, err := os.Getwd()
+		if err != nil {
+			return nil, err
+		}
+		*templateFn = path.Join(pwd, *templateFn)
+	}
+
+	return &config{
+		input:    *inputPath,
+		output:   *outputPath,
+		target:   *outputFn,
+		template: *templateFn,
+		image:    *containerImage,
+	}, nil
+}
+
+func (cfg config) render() error {
+	tmpDir, err := os.MkdirTemp("", "otel_semconvgen")
+	if err != nil {
+		return fmt.Errorf("unable to create temporary directory: %w", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	inputPath := path.Join(tmpDir, "input")
+	err = os.Mkdir(inputPath, 0700)
+	if err != nil {
+		return fmt.Errorf("unable to create input directory: %w", err)
+	}
+
+	outputPath := path.Join(tmpDir, "output")
+	err = os.Mkdir(outputPath, 0700)
+	if err != nil {
+		return fmt.Errorf("unable to create output directory: %w", err)
+	}
+
+	err = exec.Command("cp", "-a", cfg.input, path.Join(tmpDir, "input")).Run()
+	if err != nil {
+		return fmt.Errorf("unable to copy input to temp directory: %w", err)
+	}
+
+	err = exec.Command("cp", cfg.template, tmpDir).Run()
+	if err != nil {
+		return fmt.Errorf("unable to copy template to temp directory: %w", err)
+	}
+
+	cmd := exec.Command("docker", "run", "--rm",
+		"-v", fmt.Sprintf("%s:/data", tmpDir),
+		cfg.image,
+		"--yaml-root", path.Join("/data/input", path.Base(cfg.input)),
+		"code",
+		"--template", path.Join("/data", path.Base(cfg.template)),
+		"--output", path.Join("/data/output", path.Base(cfg.target)),
+	)
+	err = cmd.Run()
+	if err != nil {
+		return fmt.Errorf("unable to render template: %w", err)
+	}
+
+	err = exec.Command("cp", path.Join(tmpDir, "output", path.Base(cfg.target)), cfg.output).Run()
+	if err != nil {
+		return fmt.Errorf("unable to copy result to target: %w", err)
+	}
+
+	return nil
+}
+
+func findRepoRoot() (string, error) {
+	start, err := os.Getwd()
+	if err != nil {
+		return "", err
+	}
+
+	dir := start
+	for {
+		_, err := os.Stat(filepath.Join(dir, ".git"))
+		if errors.Is(err, os.ErrNotExist) {
+			dir = filepath.Dir(dir)
+			// From https://golang.org/pkg/path/filepath/#Dir:
+			// The returned path does not end in a separator unless it is the root directory.
+			if strings.HasSuffix(dir, string(filepath.Separator)) {
+				return "", fmt.Errorf("unable to find git repository enclosing working dir %s", start)
+			}
+			continue
+		}
+
+		if err != nil {
+			return "", err
+		}
+
+		return dir, nil
+	}
+}
```

Comment:
That is understandable, and while I don't like it, I think it makes sense.

For tests, I don't think this addresses the automation of generating these at release.  That is where I would expect the "test" to live.  And it would be much like the mod check from `check-clean-work-tree` in our CI.

To resolve this could we either add to our documentation (if we have it) what to do with this tool as part of a release, or start a ticket to create that document and include it as one of the action items in that?

Reply:
I've added some notes to the `RELEASING.md` to provide guidance on when and how to use it.  I think as we  work through #1890 and how we want to handle versioning of the generated semantic conventions the docs will need to be updated, but it should be a good start.

Reply:
I've created #1907 to track figuring out how to automate more of this process.